kube-runtime/src/watcher.rs

```diff
@@ -301,6 +327,15 @@ impl Config {
         self.list_semantic(ListSemantic::Any)
     }
 
+    /// Sets list semantic to `Stream` to make use of watch bookmarks
+    #[must_use]
+    pub fn watch_list_semantic(mut self) -> Self {
+        self.bookmarks = true;
+        self.watch_semantic = WatchSemantic::Watch;
+        self.page_size = None;
```

Comment:
```suggestion
    pub fn streaming_lists(mut self) -> Self {
        self.bookmarks = true; // streaming lists require bookmarks
        self.streaming = WatcherMode::StreamingList;
        self.page_size = None; // page size does not make sense when streaming
```

..it might also be worth setting
`self.list_semantic(ListSemantic::Any)` in this fn. it almost seems like the [streaming list docs](https://kubernetes.io/docs/reference/using-api/api-concepts/#streaming-lists) are recommending it, and it might be a good opt-in point for users (but not sure about this)

Reply:
Also, I'd probably move `page_size` into `WatcherMode`, but I suppose our hands are tied here...

Reply:
Third, `streaming` makes it sound like it should be a boolean, how about `watcher_mode` (like the enum)?

Reply:
My original advice above was indeed misguided. However, one point; you do need to set the `version_match` parameter for `WatchParams` in this mode or you'll literally get a `Error: failed to start watching object: Failed to build request: failed to validate request: WatchParams::version_match must be set to NotOlderThan when using send_initial_events` if you don't.

We don't really have much chance as to what to set the resource version to (it's either "0" or an explicit from previous stored version) so it means that it's a mistake to take [list-semantics](https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-get-and-list) and try to re-use them for [streaming-list semantics](https://kubernetes.io/docs/reference/using-api/api-concepts/#streaming-lists) because the latter only supports `NotOlderThan` semantics. The `ListSemantic` option was only ever used to produce `ListParams` so we shouldn't use it to create `WatchParams`.

If we are in `streaming_list` mode, then we must set `WatchParams::version_match` to `NotOlderThan`, I don't think there are any other ways to shoehorning this in.

Reply:
As for the `page_size.is_some() && WatchSemantic::InitialEvents` scenario, we could maybe add a `Config::validate` method that is called by the watcher before starting? That is inline with our other *Params structs.